src/ngraph/frontend/onnx_import/op/gemm.cpp

```diff
@@ -59,26 +59,31 @@ namespace ngraph
                     // code from python not implemented in c++ yet.
                     // reshape_for_matmul(node, input_a, input_b);
 
+                    // A' * B'
                     std::shared_ptr<ngraph::Node> a_dot_b =
                         std::make_shared<ngraph::op::Dot>(input_a, input_b);
 
+                    // alpha
                     std::shared_ptr<ngraph::Node> alpha_node =
                         std::make_shared<ngraph::op::Constant>(a_dot_b->get_element_type(),
                                                                ngraph::Shape{},
                                                                std::vector<double>{alpha});
                     alpha_node = make_broadcast_node(alpha_node, a_dot_b->get_shape());
+                    // alpha * A' * B'
                     a_dot_b = std::make_shared<ngraph::op::Multiply>(alpha_node, a_dot_b);
 
+                    // beta * C
                     std::shared_ptr<ngraph::Node> beta_node =
                         std::make_shared<ngraph::op::Constant>(input_c->get_element_type(),
                                                                ngraph::Shape{},
                                                                std::vector<double>{beta});
                     beta_node = make_broadcast_node(beta_node, input_c->get_shape());
-
                     input_c = std::make_shared<ngraph::op::Multiply>(beta_node, input_c);
-                    input_c = make_broadcast_node(input_c, a_dot_b->get_shape());
 
-                    return {std::make_shared<ngraph::op::Add>(a_dot_b, input_c)};
+                    // alpha * A' * B' + beta * C
+                    NodeVector broadcasted_nodes =
+                        numpy_style_broadcast_for_binary_operation(a_dot_b, input_c);
+                    return {std::make_shared<ngraph::op::Add>(a_dot_b, broadcasted_nodes.at(1))};
```

Comment:
Why are you using `a_dot_b` here instead of `broadcasted_nodes.at(0)`?
Please add a comment to explain this.